src/vs/workbench/api/electron-browser/mainThreadLanguages.ts

```diff
@@ -22,4 +26,12 @@ export class MainThreadLanguages extends MainThreadLanguagesShape {
 	$getLanguages(): TPromise<string[]> {
 		return TPromise.as(this._modeService.getRegisteredModes());
 	}
+
+	$getLanguage(resource: URI, position?: IPosition): TPromise<string> {
+		const model = this.modelService.getModel(resource);
+		if (model) {
+			return TPromise.as(position ? this._modeService.getLanguageIdentifier(model.getLanguageIdAtPosition(position.lineNumber, position.column)).language : model.getLanguageIdentifier().language);
+		}
+		return TPromise.as(this._modeService.getModeIdByFilenameOrFirstLine(resource.fsPath));
```

Comment:
Today we only peek into the first line if the file is opened (aka there is a model) or we are actually opening the model and have the contents already in hand (e.g. when opening an editor). I would say that in 99% of the cases we do it right but there are cases where we simply ignore the "firstLine" because we do not check for the file contents.